packages/pigeon/lib/generator_tools.dart

```diff
@@ -249,3 +249,42 @@ Map<String, Object> mergeMaps(
   }
   return result;
 }
+
+/// A class name that is enumerated.
+class EnumeratedClass {
+  /// Constructor.
+  EnumeratedClass(this.name, this.enumeration);
+
+  /// The name of the class.
+  final String name;
+
+  /// The enumeration of the class.
+  final int enumeration;
+}
+
+/// Custom codec's custom types are enumerated from 127 down to this number to
```

Comment:
Why down?

Reply:
It's more resistant to future changes in the StandardMessageCodec.  If we add a field to the StandardMessageCodec and someone starts using an old version of Pigeon with it, a runtime error is less likely: P(add field to StandardMessageCodec) * P(user has more than 117 referenced datatypes in an API).

Reply:
This changed now that I'm in the 128-255 range.  Now it's pretty arbitrary that I'm guaranteed not to collide.